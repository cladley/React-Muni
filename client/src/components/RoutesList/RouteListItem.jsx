import React, {Component} from 'react';

class RouteListItem extends Component {
  render() {
    const {tag, title, color} = this.props;

    return (
      <div className="route-list-item" onClick={() => this.props.onRouteClick(tag)}>
        <span className="route-list-item__tag">{tag}</span>
        <span className="route-list-item__title">{title}</span>
      </div>
    );
  }
}

export default RouteListItem;
