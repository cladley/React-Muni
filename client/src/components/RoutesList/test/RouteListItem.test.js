import React from 'react';
import {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RouteListItem from '../RouteListItem';
import renderer from 'react-test-renderer';

describe('<RouteListItem />', () => {
  const props = {
    tag: 'C',
    title: 'California',
    color: 'fffff'
  };

  it('should render correctly', () => {
    const wrapper = renderer.create(<RouteListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('user clicks item', () => {
    let wrapper;
    let onClicked = jest.fn();

    beforeEach(() => {
      onClicked.mockClear();
      wrapper = shallow(<RouteListItem {...props} onRouteClick={onClicked} />);
      wrapper.simulate('click');
    });

    it('should call prop onRouteClick()', () => {
      expect(onClicked.mock.calls.length).toBe(1);
    });

    it('should call prop onRouteClick() with correct tag', () => {
      expect(onClicked.mock.calls[0][0]).toBe('C');
    });
  });
});
