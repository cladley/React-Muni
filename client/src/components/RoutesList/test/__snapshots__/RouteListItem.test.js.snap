// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteListItem /> should render correctly 1`] = `
<div
  className="route-list-item"
  onClick={[Function]}
>
  <span
    className="route-list-item__tag"
  >
    C
  </span>
  <span
    className="route-list-item__title"
  >
    California
  </span>
</div>
`;
